### DISCUSSION TITLE
How to change "SLOPE_DATA" and other hard coded parameters

### DISCUSSION THREAD

**Discussion by psbecker** [NCAR/hrldas#223] [Q&A] [answered]
I'm trying to understand how I would change certain parameters listed in the NoahMPTable.TBL that don't seem to be listed in the input.dat or namelist files. For example, I see in the Bondville.dat there is a line called "soil_category" which i assume corresponds with the "SoilType" in the parameter table. But I'm interested in changing the SLOPE_DATA but this has only 9 options (as opposed to the 19 for SoilType) and I don't see a clear place in the input file that matches this.
Additionally, if I wanted to change other hard coded parameters like "REFKDT_DATA", how would I go about that? I don't see where in the code the parameter table is called.

Thank you

**Comment by dmocko** [comment, marked as answer]
Hi @psbecker 

In older versions of Noah, there was a 1-degree "slope type index" map with 9 slope types.

However, in Noah-MP, the "slope type" is hard-coded:

https://github.com/NCAR/noahmp/blob/c7a04d8c6d4b9533a6b99c72229ed9279fe39e46/drivers/hrldas/NoahmpIOVarInitMod.F90#L839

The reason that it is hard-coded is that the 1-degree slope type index map would cause blockiness in the lower layers
of soil moisture, due to baseflow varying by the slope type.

I don't think that the SLOPE_DATA is even read in anymore.

By the way, here is an explanation of the slope types used in Noah LSM:

https://github.com/NASA-LIS/LISF/blob/master/lis/surfacemodels/land/noah.3.9/phys/module_sf_noah39lsm.F90#L2428-L2442

**Comment by tslin2** [comment]
Hi,
https://github.com/NCAR/noahmp/blob/c7a04d8c6d4b9533a6b99c72229ed9279fe39e46/drivers/hrldas/ConfigVarInTransferMod.F90#L93
https://github.com/NCAR/noahmp/blob/c7a04d8c6d4b9533a6b99c72229ed9279fe39e46/drivers/hrldas/WaterVarInTransferMod.F90#L148

Currently, it only use the first value from the NoahmpTable.TBL
https://github.com/NCAR/noahmp/blob/c7a04d8c6d4b9533a6b99c72229ed9279fe39e46/parameters/NoahmpTable.TBL#L726
